packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -584,15 +648,35 @@ class ErrorHandlingProcessManager extends ProcessManager {
     Encoding stdoutEncoding = io.systemEncoding,
     Encoding stderrEncoding = io.systemEncoding,
   }) {
-    return _runSync(() => _delegate.runSync(
-      command,
-      workingDirectory: workingDirectory,
-      environment: environment,
-      includeParentEnvironment: includeParentEnvironment,
-      runInShell: runInShell,
-      stdoutEncoding: stdoutEncoding,
-      stderrEncoding: stderrEncoding,
-    ), platform: _platform);
+    return _runSync(() {
+      return _processRunSync(
+        _getCommandPath(command, workingDirectory),
+        _getArguments(command),
+        workingDirectory: workingDirectory,
+        environment: environment,
+        includeParentEnvironment: includeParentEnvironment,
+        runInShell: runInShell,
+        stdoutEncoding: stdoutEncoding,
+        stderrEncoding: stderrEncoding,
+      );
+    }, platform: _platform);
+  }
+
+  String _getCommandPath(List<dynamic> command, String workingDirectory, { bool strict = false }) {
+    final String executable = command.first.toString();
+    final Map<String, String> workingDirectoryCache = _resolvedExecutables[workingDirectory] ??= <String, String>{};
```

Comment:
This will not correctly handle the case where a new binary is installed in a higher-priority location after a location has already been resolved.

I don't think we should try to handle that case unless we need to, but it should probably be documented.